evergreen/config.yml

```diff
@@ -1322,173 +1239,142 @@ task_groups:
-  display_name: "Ubuntu 20.04 x86_64 (Clang 11 Encryption Enabled w/TSAN)"
-  run_on: ubuntu2004-small
+- name: ubuntu-encryption-tsan
+  display_name: "Ubuntu (Encryption Enabled w/TSAN)"
+  run_on: ubuntu2204-arm64-small
   expansions:
-    clang_url: "https://s3.amazonaws.com/static.realm.io/evergreen-assets/clang%2Bllvm-11.0.0-x86_64-linux-gnu-ubuntu-20.04.tar.xz"
-    cmake_url: "https://s3.amazonaws.com/static.realm.io/evergreen-assets/cmake-3.26.3-linux-x86_64.tar.gz"
-    cmake_bindir: "./cmake_binaries/bin"
     fetch_missing_dependencies: On
-    c_compiler: "./clang_binaries/bin/clang"
-    cxx_compiler: "./clang_binaries/bin/clang++"
+    c_compiler: "/opt/clang+llvm/bin/clang"
+    cxx_compiler: "/opt/clang+llvm/bin/clang++"
     run_with_encryption: On
     enable_tsan: On
     cmake_build_type: RelWithDebInfo
   tasks:
   - name: core_tests_group
-    distros:
-    - ubuntu2004-large
 
-- name: ubuntu2004-encryption-asan
-  display_name: "Ubuntu 20.04 x86_64 (Clang 11 Encryption Enabled w/ASAN)"
-  run_on: ubuntu2004-small
+- name: ubuntu-encryption-asan
+  display_name: "Ubuntu (Encryption Enabled w/ASAN)"
+  run_on: ubuntu2204-arm64-large
   expansions:
-    clang_url: "https://s3.amazonaws.com/static.realm.io/evergreen-assets/clang%2Bllvm-11.0.0-x86_64-linux-gnu-ubuntu-20.04.tar.xz"
-    cmake_url: "https://s3.amazonaws.com/static.realm.io/evergreen-assets/cmake-3.26.3-linux-x86_64.tar.gz"
-    cmake_bindir: "./cmake_binaries/bin"
     fetch_missing_dependencies: On
-    c_compiler: "./clang_binaries/bin/clang"
-    cxx_compiler: "./clang_binaries/bin/clang++"
-    llvm_symbolizer: "./clang_binaries/bin/llvm-symbolizer"
+    c_compiler: "/opt/clang+llvm/bin/clang"
+    cxx_compiler: "/opt/clang+llvm/bin/clang++"
     run_with_encryption: On
     enable_asan: On
     cmake_build_type: RelWithDebInfo
   tasks:
   - name: core_tests_group
-    distros:
-    - ubuntu2004-large
 
-- name: ubuntu2004-release
-  display_name: "Ubuntu 20.04 x86_64 (Clang 11 Release build)"
-  run_on: ubuntu2004-small
+- name: ubuntu-release
+  display_name: "Ubuntu (Release build)"
+  run_on: ubuntu2204-arm64-large
   expansions:
-    clang_url: "https://s3.amazonaws.com/static.realm.io/evergreen-assets/clang%2Bllvm-11.0.0-x86_64-linux-gnu-ubuntu-20.04.tar.xz"
-    cmake_url: "https://s3.amazonaws.com/static.realm.io/evergreen-assets/cmake-3.26.3-linux-x86_64.tar.gz"
-    bloaty_url: "https://s3.amazonaws.com/static.realm.io/evergreen-assets/bloaty-v1.1-39-gefc1c61-ubuntu2004-x86_64.tar.gz"
-    cmake_bindir: "./cmake_binaries/bin"
     cmake_build_type: RelWithDebInfo
     fetch_missing_dependencies: On
-    c_compiler: "./clang_binaries/bin/clang"
-    cxx_compiler: "./clang_binaries/bin/clang++"
+    c_compiler: "/opt/clang+llvm/bin/clang"
+    cxx_compiler: "/opt/clang+llvm/bin/clang++"
     python3: /opt/mongodbtoolchain/v3/bin/python3
   tasks:
   - name: compile_test_and_package
-    distros:
-    - ubuntu2004-large
   - name: benchmarks
```

Comment:
this will effectively break all recorded history for our benchmarks tracked by performance regression service, no?

Maybe we can keep old builders just for benchmarks for a while until we collect at least a few months of datapoints?

Current dashboard for realm-core tracks benchmarks from ubuntu2004-release, ubuntu2004-arm64, macos-release and macos-1100-arm64-release

Reply:
it won't remove all recorded history but it will start a new history. i'm fine with keeping the names of these build variants the same for now (i.e. the variant name will be ubuntu2004-release, but it's really ubuntu 22.04 on arm64). or we could keep the older build variants around but only run the benchmarks. or we could just figure out how to add the old variants as a point of comparison in the dashboard. i asked if there was a way to rename the variants on the perf side but no luck.

Reply:
okay, I can keep the old build variants around for a while and just run benchmarks on them. we also do have an updated x86_64 clang ubuntu in this PR, so I could also do the OS/compiler update but keep running benchmarks on x86_64 for now.

Reply:
As for abandoning x86_64, i'm going to keeping x86_64 linux builder around - in this PR its the code coverage builder - but I do think x86_64 shouldn't be the "default" type of builder going forward. Like if we need to test with geospatial disabled or multiplexing disabled or with a fancy sanitizer, we don't need to pay the cost of x86_64.

Reply:
@kiburtse, I've added the old builders back with just benchmarks. I've also added a clang 18 x86_64 benchmark builder so we can have an apples-to-apples comparison between the old clang benchmarks and the new clang benchmarks without also adding the arm64 variable. let me know if this works for you.